### PR TITLE
GLK:SCOTT: Fix assert failed after calling Scott::fatal()

### PR DESCRIPTION
**Pull request by hari01584**
As raised by [14257](https://bugs.scummvm.org/ticket/14257), Scott engine gives error of 
`
scummvm: ./common/array.h:234: T& Common::Array<T>::operator[](Common::Array<T>::size_type) [with T = Glk::Scott::Item; Common::Array<T>::size_type = unsigned int]: Assertion idx < _size' failed.
`

It happens because the shouldQuit() function https://github.com/scummvm/scummvm/blob/c7c2141d0a88b194e8981cc16456fac9f8b2c2ce/engines/glk/scott/scott.cpp#L145 is checked after saving state [_G(_initialState) = saveCurrentState();](https://github.com/scummvm/scummvm/blob/c7c2141d0a88b194e8981cc16456fac9f8b2c2ce/engines/glk/scott/scott.cpp#L143)  which obviously in unsupported game will crash!

Fixed it by checking if there is quit event when calling saveCurrentState()!

### CODE REVIEW

**Comment by angstsmurf** [comment]
Can confirm that this makes ScummVM exit cleanly instead of crashing with an assert when trying to run an unsupported game.

File: engines/glk/scott/scott.cpp [hunk 1]
```diff
@@ -140,7 +140,8 @@ void Scott::runGame() {
 	// Check for savegame
 	_saveSlot = ConfMan.hasKey("save_slot") ? ConfMan.getInt("save_slot") : -1;
 
-	_G(_initialState) = saveCurrentState();
+	if(!shouldQuit())
```
Comment (sev-) [inline engines/glk/scott/scott.cpp:143]: Please fix the formatting here: `if (!shouldQuit())`. 

Our formatting conventions can be found here: https://wiki.scummvm.org/index.php?title=Code_Formatting_Conventions

Comment (hari01584) [reply engines/glk/scott/scott.cpp]: Done!

**Comment by sev-** [comment]
Thank you

**Pull request merged**